internal/datacoord/compaction_task_clustering.go

```diff
@@ -343,11 +343,18 @@ func (t *clusteringCompactionTask) resetSegmentCompacting() {
 
 func (t *clusteringCompactionTask) processFailedOrTimeout() error {
 	log.Info("clean task", zap.Int64("triggerID", t.GetTriggerID()), zap.Int64("planID", t.GetPlanID()), zap.String("state", t.GetState().String()))
-	// revert segment level
+	// revert segments meta
 	var operators []UpdateOperator
+	// revert level of input segments
+	// L1 : L1 ->(processPipelining)-> L2 ->(processFailedOrTimeout)-> L1
+	// L2 : L2 ->(processPipelining)-> L2 ->(processFailedOrTimeout)-> L2
 	for _, segID := range t.InputSegments {
```

Comment:
A little bit confused, 
Why input segments will be marked as L2 level in the process of Executing. Will this make these L1 segments invisible in the process of clustering compaction?

Reply:
Actually it make a strong protection mix compaction won't include these L1 segments. It is still visible. Yes, I agree it is a little confused.